indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -445,6 +468,87 @@ private void extractDateCreated(Element mods, IndexDocumentBean idb) {
         }
     }
 
+    private void extractRights(Element mods, IndexDocumentBean idb) {
+        List<String> rights = new ArrayList<>();
+        List<String> rightsOaiPmh = new ArrayList<>();
+        List<String> rightsUri = new ArrayList<>();
+
+        List<Element> rightsEls = mods.getChildren("accessCondition", JDOMNamespaceUtil.MODS_V3_NS);
+        for (Element rightsEl : rightsEls) {
+            String accessType = rightsEl.getAttributeValue("type");
+            if (accessType != null && accessType.trim().equalsIgnoreCase("use and reproduction")) {
+                String href = rightsEl.getAttributeValue("href", JDOMNamespaceUtil.XLINK_NS);
+                String modsRightsStatement = rightsEl.getTextTrim();
+                boolean hasUrl = !StringUtils.isBlank(href);
+                boolean hasRightsText = !StringUtils.isBlank(modsRightsStatement);
+
+                if (!hasRightsText && !hasUrl) {
+                    continue;
+                }
+
+                // No vocabulary. Add MODS text
+                if (hasRightsText && !hasUrl) {
+                    rights.add(modsRightsStatement);
+                }
+
+                if (hasUrl && !hasRightsText) {
+                    String trimmedHref = href.trim();
+                    if (rightsUriMap.containsKey(trimmedHref)) {
+                        String mappedRightsText = rightsUriMap.get(trimmedHref);
+                        rightsOaiPmh.add(setUrlProtocol(trimmedHref));
+                        rightsOaiPmh.add(mappedRightsText);
+                        rights.add(mappedRightsText);
+                    }
+                    rightsUri.add(trimmedHref);
+                }
+
+                if (hasRightsText && hasUrl) {
```

Comment:
I think we can merge the logic for `hasUrl && !hasRightsText` with this, might even be able to use this logic for all 3 cases:

1. if there is a url, check if its in vocab:
    1. if it is:
        1. add it to rightsUri
        2. set indexingText to the display label from vocab
        3. add the transformed version to rightsOaiPmh
    2. if it was not in the vocab, log a warning and discard the URI
2. if indexingText == null
    1. then, set it to the text value from the MODS field
    2. else, compare it to the value from the MODS field, and log error if they don't match
3. if indexingText != null
    1. add indexingText to `rights` and `rightsOaiPmh`